x-pack/solutions/chat/plugins/workchat-app/common/chat_events.ts

```diff
@@ -28,6 +28,12 @@ export interface ToolResultEvent extends ChatEventBase {
   };
 }
 
+export interface ProgressionEvent extends ChatEventBase {
+  type: 'progression';
+  progressionType: string;
+  data: Record<string, any>;
+}
```

Comment:
So, this will likely be refactored later in something more robust and having more meta (e.g. source node and so on), but this is "good enough" for now.